### ISSUE TITLE
LLAMA 2 HF tokenizer len is 32001

### ISSUE THREAD

**Issue by ari9dam** [huggingface/transformers#24899]
### System Info

Installed from source (4.32.0.dev0)

```
from transformers import AutoTokenizer, AutoModelForCausalLM

tokenizer = AutoTokenizer.from_pretrained("meta-llama/Llama-2-13b-hf")
model = AutoModelForCausalLM.from_pretrained("meta-llama/Llama-2-13b-hf")
print(len(tokenizer)) #32001
print(model.config.vocab_size) #32000
```

### Information

- [ ] The official example scripts
- [ ] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

```
from transformers import AutoTokenizer, AutoModelForCausalLM

tokenizer = AutoTokenizer.from_pretrained("meta-llama/Llama-2-13b-hf")
model = AutoModelForCausalLM.from_pretrained("meta-llama/Llama-2-13b-hf")
print(len(tokenizer)) #32001
print(model.config.vocab_size) #32000
```

### Expected behavior

model vocab size and tokenizer len should be 32000. It seems the padding token of the tokenizer is set to '\<unk\>'. Which is not normally the case. It's normally not set.

**Comment by sgugger** [comment]
cc @ArthurZucker

**Comment by ArthurZucker** [comment]
Hey! Yes this is not entirely expected, we update the slow tokenizer, but the fast version did not get the update. I'll open PRs to fix this!

**Comment by psinger** [comment]
@ArthurZucker jfyi I see the same behavior for slow and fast on 4.31.0

**Comment by ari9dam** [comment]
Is there any plan to fix it?

**Comment by ArthurZucker** [comment]
It is fixed on all model!

**Comment by es94129** [comment]
@ArthurZucker  Does the fix really resolves this? I pip installed `transformers` from Github, and there is still a mismatch.

```python
from transformers import AutoTokenizer, AutoModelForCausalLM

# Using fast tokenizer by default
tokenizer = AutoTokenizer.from_pretrained("meta-llama/Llama-2-7b-chat-hf")
model = AutoModelForCausalLM.from_pretrained("meta-llama/Llama-2-7b-chat-hf")
print(len(tokenizer)) # 32001
print(model.config.vocab_size) # 32000
print(tokenizer.get_added_vocab()) # {'<pad>': 32000}
```

Transformers version: `4.32.0.dev0`

I'm suspecting the issue is intended by Llamav2 authors: https://huggingface.co/docs/transformers/main/model_doc/llama2
> The original model uses pad_id = -1 which means that there is no padding token. We can’t have the same logic, make sure to add a padding token using tokenizer.add_special_tokens({"pad_token":"<pad>"}) and resize the token embedding accordingly. You should also set the model.config.pad_token_id. The embed_tokens layer of the model is initialized with self.embed_tokens = nn.Embedding(config.vocab_size, config.hidden_size, self.config.padding_idx), which makes sure that encoding the padding token will output zeros, so passing it when initializing is recommended.

**Comment by ArthurZucker** [comment]
Will check asap! Might be the fast tokenizer that did not get updated.

**Comment by ArthurZucker** [comment]
Sorry but no, I cannot reproduce your issue. I do not know if you maybe did not update cached files but I have this: 
```python 
>>> from transformers import AutoTokenizer
>>> tokenizer = AutoTokenizer.from_pretrained("meta-llama/Llama-2-7b-chat-hf")
>>> len(tokenizer)
32000
>>> print(tokenizer.get_added_vocab())
{}
```

**Comment by es94129** [comment]
Thanks for confirming. You're right, it's because of the locally cached files. Downloading them from huggingface and re-running the code now works fine.
